server/api/api.go

```diff
@@ -38,6 +39,11 @@ func Init(h *httputils.Handler, env mscalendar.Env, notificationProcessor mscale
 
 	dialogRouter := h.Router.PathPrefix(config.PathAutocomplete).Subrouter()
 	dialogRouter.HandleFunc(config.PathUsers, api.autocompleteUsers)
+	dialogRouter.HandleFunc(config.PathChannels, api.autocompleteChannels)
```

Comment:
Would this take care of excluding channels where the user can't write?

Reply:
By itself, I don't think so. In the core webapp we usually check that at runtime by looking at the channel membership, which should all be loaded into the webapp on page load. 

https://github.com/mattermost/mattermost/blob/master/webapp/channels/src/components/advanced_create_comment/index.ts#L81

```ts
import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
import Permissions from 'mattermost-redux/constants/permissions';

if (haveIChannelPermission(state, teamId, channelId, Permissions.CREATE_POST)
```

Maybe we keep it as-is, because it works. Downside is the current implementation is reinventing the wheel as far as search goes. The plugin is currently doing a search on all channels on the server side, and then filtering ones related to the user afterwards. I think I would prefer the above solution, to keep some stuff out of the plugin's server part.